### ISSUE TITLE
Tf-gpu training with train.py maxes out RAM & CPU and crashes

### ISSUE THREAD

**Issue by csindic** [tensorflow/models#4017]
So I am experiencing an odd combination of (mostly unresolved) issues others have experienced. I should note that this is a unique issue in that I have not seen anyone experience the same amalgamation of these issues at once. I am (unfortunately) running Windows 10 with anaconda and pip for most of my modules. I have tried all combinations of CUDA toolkit and CuDNN (versions, that is). 

Python 3.6
Have used all combinations of CuDNN 7,7.1,8 and CUDA 8, 9, 9.1 & tf-gpu from conda, pip etc.

p5000 16GB
Dual Xeon Gold 
64GB RAM

When I run the standard "python train.py --logtostderr --train_dir=training/ --pipeline_config_path=training/ssd_mobilenet_v1_pets.config" I get the following.

WARNING:tensorflow:From C:\Users\caleb\AppData\Local\Continuum\anaconda3\envs\gpu2\lib\site-packages\tensorflow\contrib\learn\python\learn\datasets\base.py:198: retry (from tensorflow.contrib.learn.python.learn.datasets.base) is deprecated and will be removed in a future version.
Instructions for updating:
Use the retry module or similar alternatives.
WARNING:tensorflow:From C:\Users\caleb\AppData\Local\Continuum\anaconda3\envs\gpu2\lib\site-packages\object_detection\trainer.py:228: create_global_step (from tensorflow.contrib.framework.python.ops.variables) is deprecated and will be removed in a future version.
Instructions for updating:
Please switch to tf.train.create_global_step
INFO:tensorflow:depth of additional conv before box predictor: 0
INFO:tensorflow:depth of additional conv before box predictor: 0
INFO:tensorflow:depth of additional conv before box predictor: 0
INFO:tensorflow:depth of additional conv before box predictor: 0
INFO:tensorflow:depth of additional conv before box predictor: 0
INFO:tensorflow:depth of additional conv before box predictor: 0
WARNING:root:Variable [FeatureExtractor/MobilenetV1/Conv2d_0/BatchNorm/beta/ExponentialMovingAverage] is not available in checkpoint
* * *
*A LOT more of the same warning for different items (~100)*
* * *
WARNING:root:Variable [FeatureExtractor/MobilenetV1/Conv2d_9_pointwise/weights/RMSProp_1] is not available in checkpoint
WARNING:tensorflow:From C:\Users\caleb\AppData\Local\Continuum\anaconda3\envs\gpu2\lib\site-packages\tensorflow\contrib\slim\python\slim\learning.py:736: Supervisor.__init__ (from tensorflow.python.training.supervisor) is deprecated and will be removed in a future version.
Instructions for updating:
Please switch to tf.train.MonitoredTrainingSession
WARNING:tensorflow:From C:\Users\caleb\AppData\Local\Continuum\anaconda3\envs\gpu2\lib\site-packages\tensorflow\contrib\slim\python\slim\learning.py:736: Supervisor.__init__ (from tensorflow.python.training.supervisor) is deprecated and will be removed in a future version.
Instructions for updating:
Please switch to tf.train.MonitoredTrainingSession
2018-04-17 21:32:14.570705: I T:\src\github\tensorflow\tensorflow\core\platform\cpu_feature_guard.cc:140] Your CPU supports instructions that this TensorFlow binary was not compiled to use: AVX2
2018-04-17 21:32:14.978433: I T:\src\github\tensorflow\tensorflow\core\common_runtime\gpu\gpu_device.cc:1344] Found device 0 with properties:
name: Quadro P5000 major: 6 minor: 1 memoryClockRate(GHz): 1.7335
pciBusID: 0000:73:00.0
totalMemory: 16.00GiB freeMemory: 13.38GiB
2018-04-17 21:32:15.000099: I T:\src\github\tensorflow\tensorflow\core\common_runtime\gpu\gpu_device.cc:1423] Adding visible gpu devices: 0
2018-04-17 21:33:14.097334: I T:\src\github\tensorflow\tensorflow\core\common_runtime\gpu\gpu_device.cc:911] Device interconnect StreamExecutor with strength 1 edge matrix:
2018-04-17 21:33:14.110286: I T:\src\github\tensorflow\tensorflow\core\common_runtime\gpu\gpu_device.cc:917]      0
2018-04-17 21:33:14.115284: I T:\src\github\tensorflow\tensorflow\core\common_runtime\gpu\gpu_device.cc:930] 0:   N
2018-04-17 21:33:14.120510: I T:\src\github\tensorflow\tensorflow\core\common_runtime\gpu\gpu_device.cc:1041] Created TensorFlow device (/job:localhost/replica:0/task:0/device:GPU:0 with 12974 MB memory) -> physical GPU (device: 0, name: Quadro P5000, pci bus id: 0000:73:00.0, compute capability: 6.1)
INFO:tensorflow:Restoring parameters from ssd_mobilenet_v1_coco_11_06_2017/model.ckpt
INFO:tensorflow:Restoring parameters from ssd_mobilenet_v1_coco_11_06_2017/model.ckpt
INFO:tensorflow:Running local_init_op.
INFO:tensorflow:Running local_init_op.
INFO:tensorflow:Done running local_init_op.
INFO:tensorflow:Done running local_init_op.
INFO:tensorflow:Starting Session.
INFO:tensorflow:Starting Session.
INFO:tensorflow:Saving checkpoint to path training/model.ckpt
INFO:tensorflow:Saving checkpoint to path training/model.ckpt
INFO:tensorflow:Starting Queues.
INFO:tensorflow:Starting Queues.
* * *
Somewhere around this point my GPU memory shoots up to about %80 (expected), but then CPU utilization RAPIDLY climbs to %100 and my RAM usage SLOWLY begins to climb.
* * *
INFO:tensorflow:global_step/sec: 0
INFO:tensorflow:global_step/sec: 0
INFO:tensorflow:global_step/sec: 0
INFO:tensorflow:global_step/sec: 0
* * *
Upon RAM usage hitting near %100,  python crashes and the following displays:
* * *
2018-04-17 21:36:12.158184: E T:\src\github\tensorflow\tensorflow\stream_executor\cuda\cuda_dnn.cc:396] Loaded runtime CuDNN library: 7103 (compatibility version 7100) but source was compiled with 7003 (compatibility version 7000).  If using a binary install, upgrade your CuDNN library to match.  If building from sources, make sure the library loaded at runtime matches a compatible version specified during compile configuration.
2018-04-17 21:36:12.189298: F T:\src\github\tensorflow\tensorflow\core\kernels\conv_ops.cc:712] Check failed: stream->parent()->GetConvolveAlgorithms( conv_parameters.ShouldIncludeWinogradNonfusedAlgo<T>(), &algorithms)`

**Comment by derekjchow** [comment]
Just to clarify, are you using the standard pets config and dataset?

**Comment by csindic** [comment]
Yes, ssd_mobilenet_v1_pets.config. 
Additionally, I have successfully used this dataset before but not on _this_ computer with GPU.
 It works using CPU on this (new) machine, as well as the GPU and CPU on my laptop in both linux and windows.

**Comment by csindic** [comment]
Solved it! After over 10 hours, finally! 

I had several versions of tf-gpu in my site-packages dir for anaconda **as a whole**, rather than the individual env I was using. I deleted these and ran pip install tf-nightly-gpu. I am using CUDA 9.0 and CuDNN 7.0 or 7.1 (I can't see which one in the documentation and I don't want to experiment because I am training right now).
